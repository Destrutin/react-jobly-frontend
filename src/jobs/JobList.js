import React, {useEffect, useState} from "react";
import SearchBar from "../search/SearchBar";
import Api from "../api/Api";
import JobCardList from "./JobCardList";

function JobList() {
    const [jobs, setJobs] = useState(null);

    useEffect(() => {
        async function fetchData() {
            try {
                let jobs = await Api.getJobs();
                setJobs(jobs);
            } catch (err) {
                console.error("Error fetching data", err);
            }
        }
        fetchData();
    }, []);

    async function search(title) {
        let jobs = await Api.getJobs(title);
        setJobs(jobs);
    }

    return (
        <div className="JobList">
            <SearchBar search={search}/>
                <JobCardList jobs={jobs}/>
        </div>
    )
}

export default JobList;