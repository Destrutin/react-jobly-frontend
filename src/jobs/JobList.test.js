import React from 'react';
import {render} from '@testing-library/react';
import JobList from './JobList';

test('renders job list without crashing', () => {
    render(
        <JobList/>
    )
})

test('matches snapshot', () => {
    const {asFragment} = render(
        <JobList/>
    )
    expect(asFragment()).toMatchSnapshot();
})