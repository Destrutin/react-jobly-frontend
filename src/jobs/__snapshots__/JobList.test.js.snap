// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="JobList"
  >
    <div
      class="SearchBar"
    >
      <form>
        <input
          name="searchTerm"
          type="text"
          value=""
        />
        <button
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
    <div
      class="JobCardList"
    />
  </div>
</DocumentFragment>
`;
